ui/shared/GasSelector.qml

```diff
@@ -0,0 +1,206 @@
+            anchors.top: gasSlider.bottom
+            anchors.topMargin: Style.current.padding
+            anchors.left: parent.left
+            text: qsTr("Slow")
+            font.pixelSize: 15
+            color: Style.current.textColor
+            visible: parent.visible
+        }
+
+        StyledText {
+            id: labelOptimal
+            anchors.top: gasSlider.bottom
+            anchors.topMargin: Style.current.padding
+            anchors.horizontalCenter: gasSlider.horizontalCenter
+            text: qsTr("Optimal")
+            font.pixelSize: 15
+            color: Style.current.textColor
+            visible: parent.visible
+        }
+
+        StyledText {
+            id: labelFast
+            anchors.top: gasSlider.bottom
+            anchors.topMargin: Style.current.padding
+            anchors.right: parent.right
+            text: qsTr("Fast")
+            font.pixelSize: 15
+            color: Style.current.textColor
+            visible: parent.visible
+        }
+    }
+
+    StyledButton {
+        id: buttonReset
+        anchors.top: sliderWrapper.bottom
+        anchors.topMargin: sliderWrapper.visible ? Style.current.smallPadding : 0
+        anchors.right: buttonAdvanced.left
+        anchors.rightMargin: -Style.current.padding
+        label: qsTr("Reset")
+        btnColor: "transparent"
+        labelFontSize: 13
+        visible: !sliderWrapper.visible
+        onClicked: {
+            gasSlider.value = root.defaultGasPrice()
+            inputGasPrice.text = root.defaultGasPrice()
+        }
+    }
+
+    StyledButton {
+        id: buttonAdvanced
+        anchors.top: sliderWrapper.bottom
+        anchors.topMargin: sliderWrapper.visible ? Style.current.smallPadding : 0
+        anchors.right: parent.right
+        anchors.rightMargin: -Style.current.padding
+        label: qsTr("Advanced")
+        btnColor: "transparent"
+        labelFontSize: 13
+        onClicked: {
+            customNetworkFeeDialog.open()
+        }
+    }
+
+    ModalPopup {
+        id: customNetworkFeeDialog
+        title: qsTr("Custom Network Fee")
+        height: 386
+
+        Input {
+          id: inputGasLimit
+          label: qsTr("Gas limit")
+          text: root.gasLimit
+          customHeight: 56
+          anchors.top: parent.top
+          onTextChanged: {
+              if (inputGasLimit.text.trim() === "") {
+                  inputGasLimit.text = root.gasLimit
+              }
+              root.updateGasEthValue()
+          }
+        }
+
+        Input {
+          id: inputGasPrice
+          label: qsTr("Gas price")
+          anchors.top: inputGasLimit.bottom
+          anchors.topMargin: Style.current.smallPadding
+          customHeight: 56
+          text: root.defaultGasPrice()
+          onTextChanged: {
+              if (inputGasPrice.text.trim() === "") {
+                  inputGasPrice.text = root.defaultGasPrice()
+              }
+              root.updateGasEthValue()
+          }
```

Comment:
Unfortunately, I didn't manage to apply custom widths or using `RowLayout` here for the `Input` components so I can make it look like on the screen designs... not sure what's going on there..

Summoning @jrainville !

Reply:
I assume it's because Input has `anchors.left` and `anchors.right` defined. If you want to use `width` you need to dismiss those `anchors` with `anchors.right: undefined`, then the width should work. Or just use the anchors and ignore the width.

Reply:
That did the trick!